File: csharp/src/JewelleryStorage.cs
using NecklaceRefactoringKata.JewelleryTypes;

namespace NecklaceRefactoringKata
{
    public class JewelleryStorage
    {
        public readonly JewelleryBox Box;
        public readonly List<JewelleryBase> Tree;
        public readonly List<JewelleryBase> TravelRoll;
        public readonly List<JewelleryBase> Safe;
        public readonly List<JewelleryBase> DresserTop;

        public JewelleryStorage()
        {
            Box = new JewelleryBox();
            Tree = new List<JewelleryBase>();
            TravelRoll = new List<JewelleryBase>();
            Safe = new List<JewelleryBase>();
            DresserTop = new List<JewelleryBase>();
        }
        public bool IsInTravelRoll(JewelleryBase item)
        {
            return TravelRoll.Contains(item);
        }
    }
}

File: php/src/JewelleryStorage.php
<?php

namespace NecklaceRefactoringKata;

class JewelleryStorage
{
    public readonly JewelleryBox $box;

    /**
     * @var Jewellery[]
     */
    public readonly array $tree;

    /**
     * @var Jewellery[]
     */
    public readonly array $travelRoll;

    /**
     * @var Jewellery[]
     */
    public readonly array $safe;

    /**
     * @var Jewellery[]
     */
    public readonly array $dresserTop;

    public function __construct()
    {
        $this->box = new JewelleryBox();
        $this->tree = [];
        $this->travelRoll = [];
        $this->safe = [];
        $this->dresserTop = [];
    }

    public function isInTravelRoll(Jewellery $item): bool
    {
        return in_array($item, $this->travelRoll);
    }
}
